### PR TITLE
Add span_kind to the honeycomb exporter

### PR DESCRIPTION
**Pull request by chris-smith-zocdoc**
@paulosman 

Adds the span kind to the event sent to Honeycomb, allowing for queries by the type of span (client / server)

Let me know if you thing I should rename the field or lowercase the values.

### CODE REVIEW

**Review by bogdandrutu — DISMISSED** [review]

**Comment by bogdandrutu** [comment]
@lizthegrey can you review this?

File: exporter/honeycombexporter/honeycomb.go
Comment (lizthegrey) [inline exporter/honeycombexporter/honeycomb.go:180]: we typically use `meta.span_type` in our taxonomy, and lowercase. see https://github.com/honeycombio/beeline-go/blob/76ad42c601cfe475c54a3c73b8ef84c2f4aabbce/trace/trace.go#L457 (cc @maplebed @paulosman)

Comment (chris-smith-zocdoc) [reply exporter/honeycombexporter/honeycomb.go:180]: `meta.span_type` is currently used for `span_event` and `link` per the metadata kind docs https://docs.honeycomb.io/working-with-your-data/managing-your-data/definitions/#metadata-kind
 
If sending values other than link/span_event is alright with you, I can change it

Comment (lizthegrey) [reply exporter/honeycombexporter/honeycomb.go:180]: We need a bit of time to think through this API/schema design, is it okay with you if we come back to you in a week with a decision? Regardless, please go ahead and lowercase the constants as we're pretty sure on that.

Comment (chris-smith-zocdoc) [reply exporter/honeycombexporter/honeycomb.go:180]: Yeah no rush on my end

Comment (chris-smith-zocdoc) [reply exporter/honeycombexporter/honeycomb.go:180]: Separate action item: I think the exporter needs to be updated to the new internal data format, which will give it access to the other span kinds (producer, consumer, internal)

**Comment by bogdandrutu** [comment]
@lizthegrey do I understand correctly to hold on this PR until next week?

**Comment by lizthegrey** [comment]
> @lizthegrey do I understand correctly to hold on this PR until next week?

Affirm, please hold until we can figure out the right schema on our side. Thanks!

**Comment by bogdandrutu** [comment]
Revoking my approval until @lizthegrey confirms the solution.

**Comment by bogdandrutu** [comment]
@lizthegrey ping

**Comment by tigrannajaryan** [comment]
@paulosman and @MikeGoldsmith do you plan to continue working on this?

**Comment by MikeGoldsmith** [comment]
Ah, this got lost in our backlog. We'll look into it ASAP.

**Comment by MikeGoldsmith** [comment]
@chris-smith-zocdoc We've finished our internal work and we've settled on `meta.annotation_type` for span events and span links.

Sorry for the delay, there was some additional work we needed to complete beforehand.

**Comment by MikeGoldsmith** [comment]
Hey @chris-smith-zocdoc, I've re-reviewed your PR and please can you make the following changes:

1) [line 61](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/474/files#diff-2697d46b3215242d73f76591d5c85a8cR61) replace spanType with annotation_type on spanEvent struct
2) [line 72](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/474/files#diff-2697d46b3215242d73f76591d5c85a8cR72) replace spanType with annotation_type on link struct
3) [link 181](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/474/files#diff-2697d46b3215242d73f76591d5c85a8cR181) use span_type as you are already, what you have already is good :+1:

Thanks!

**Comment by bogdandrutu** [comment]
Please also rebase

**Comment by MikeGoldsmith** [comment]
@chris-smith-zocdoc If you don't have time to make the changes, I can do them and the rebase. I won't be able to use this PR, but I can base a branch on your original branch so your commits are not lost.

Thanks

**Review by MikeGoldsmith — APPROVED** [review]
Looks good for me - thanks @chris-smith-zocdoc.

**Comment by chris-smith-zocdoc** [comment]
@MikeGoldsmith rebased and made changes

**Comment by MikeGoldsmith** [comment]
Thanks @chris-smith-zocdoc. We're happy with the changes above but need to finalise a couple of things on our side. Please do not merge this and I'll update when we're ready.

**Comment by bogdandrutu** [comment]
@MikeGoldsmith please mark this with "required changes" so we don't by mistake merge this

**Comment by MikeGoldsmith** [comment]
Thanks @bogdandrutu - after some internal comms we're happy for this to merged as there won't be a negative effect if this were merged / released. Happy for this to be merged now 😄

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
